### DISCUSSION TITLE
Not able to cross compile liboqs for aarch64 on ubuntu 16.04

### DISCUSSION THREAD

**Discussion by ranjankumar23** [open-quantum-safe/liboqs#1500] [Q&A]
Not able to cross compile liboqs for aarch64 on ubuntu 16.04 .
It expects aarch64-linux-gnu-gcc-7 which is not available in ubuntu 16.04.

need help.

**Comment by baentsch** [comment]
This is a pretty dated ubuntu that we're not actively supporting. What stops you from upgrading (or at least running a docker image with a more current version)?

**Reply by ranjankumar23** [reply]
Thank you @baentsch for your response. The main binary and OS where I have to integrate this liboqs and openssl has older glibc 2.26 (can compile only on 16.04)

**Reply by baentsch** [reply]
Well, then it seems you'll have to build `aarch64-linux-gnu-gcc-7` on/for that platform. Alternatively, why don't you build natively instead of cross?

**Reply by ranjankumar23** [reply]
Thank you @baentsch for your response.
I built gcc 7.5.0 and also latest gcc using following commands. 
../gcc/configure --target=aarch64-linux-gnu --prefix=/usr/local --disable-nls --enable-languages=c,c++ --without-headers
make all-gcc

liboqs cmake config is giving assembler error.
`cmake -GNinja -DCMAKE_INSTALL_PREFIX=/home/ranjan/ranjan/openssl/oqs -DCMAKE_TOOLCHAIN_FILE=../.CMake/toolchain_arm64.cmake -DBUILD_SHARED_LIBS=ON -DOQS_USE_OPENSSL=OFF ..
-- The C compiler identification is unknown
-- The ASM compiler identification is GNU
-- Found assembler: /usr/local/bin/aarch64-linux-gnu-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: /usr/local/bin/aarch64-linux-gnu-gcc
-- Check for working C compiler: /usr/local/bin/aarch64-linux-gnu-gcc - broken
CMake Error at /usr/local/share/cmake-3.18/Modules/CMakeTestCCompiler.cmake:66 (message):
  The C compiler

    "/usr/local/bin/aarch64-linux-gnu-gcc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /home/ranjan/ranjan/liboqs/build/CMakeFiles/CMakeTmp

    Run Build Command(s):/usr/bin/ninja cmTC_045c2 && [1/2] Building C object CMakeFiles/cmTC_045c2.dir/testCCompiler.c.o
    FAILED: /usr/local/bin/aarch64-linux-gnu-gcc    -o CMakeFiles/cmTC_045c2.dir/testCCompiler.c.o -c testCCompiler.c
    as: unrecognized option '-EL'
    ninja: build stopped: subcommand failed.
`
May be i am missing something in gcc compilation.

**Reply by baentsch** [reply]
> May be i am missing something in gcc compilation.

Either that or in setting up your environment. One of the first hits on SO: https://stackoverflow.com/questions/66799641/usr-bin-as-unrecognized-option-el. To me this doesn't look like an OQS problem.

**Reply by ranjankumar23** [reply]
Thank you @baentsch. Please close this issue as it is not specific to liboqs.
